File: app/src/main/java/com/instar/frontend_android/ui/activities/LoginSmsActivity.java
package com.instar.frontend_android.ui.activities;

import android.content.Context;
import android.os.Bundle;
import android.text.InputType;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.instar.frontend_android.R;
import com.instar.frontend_android.ui.customviews.ViewEditText;
import com.instar.frontend_android.ui.customviews.ViewEffect;

public class LoginSmsActivity extends AppCompatActivity {
    private ImageButton imageBack;
    private TextView btnText;

    private TextView title;
    private TextView textNote;
    private View layout;
    private View layoutSms;
    private EditText SmsText;
    private TextView labelSms;
    private ImageButton btnEyes;
    private TextView btnNewSms;
    private Button btnSms;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login_sms);
        imageBack = findViewById(R.id.imageBack);
        layout = findViewById(R.id.layout);
        layoutSms = findViewById(R.id.sms);
        SmsText = layoutSms.findViewById(R.id.editText);
        labelSms = layoutSms.findViewById(R.id.textView);
        btnEyes = layoutSms.findViewById(R.id.btnImage);
        title =  findViewById(R.id.smsTitle).findViewById(R.id.Title);
        textNote =  findViewById(R.id.smsTitle).findViewById(R.id.textNote);
        btnText = findViewById(R.id.smsTitle).findViewById(R.id.btnText);
        btnNewSms = findViewById(R.id.btnNewSms);
        btnSms = findViewById(R.id.btnSms);
        loadActivity();
        initView();
    }
    public void loadActivity() {
        title.setText("Xác nhận tài khoản");
        textNote.setText("Chúng tôi đã gửi mã đến email của bạn.Hãy nhập mã đó để xác nhận tài khoản");
        btnText.setText("Không thể xác nhận tài khoản?");
        labelSms.setText("Nhập mã");
        btnNewSms.setText("Gửi lại mã");
        SmsText.setHint("Nhập mã");
        btnSms.setText("Tiếp tục");
        SmsText.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_PASSWORD);
        btnEyes.setBackgroundResource(R.drawable.ic_instagram_eyes_off);
    }
    public void initView() {
        ViewEditText viewEditText = new ViewEditText();
        viewEditText.EditTextEyes(layoutSms,SmsText,labelSms,btnEyes);
        ViewEffect.ViewText(btnText);
        btnText.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

            }
        });
        layout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
                imm.hideSoftInputFromWindow(v.getWindowToken(), 0);
                SmsText.clearFocus();
                if (SmsText.getText().toString().isEmpty()) {
                    setSms();
                }
            }
        });
        imageBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
        effectClick();
    }

    public void setSms() {
        labelSms.setVisibility(View.GONE);
        layoutSms.setBackground(getDrawable(R.drawable.border_component_login_dow));
        SmsText.setHint("Nhập mã");
    }

    public void effectClick() {
        ViewEffect.ViewText(btnNewSms);
        ViewEffect.ImageBack(imageBack);
    }
}

File: app/src/main/java/com/instar/frontend_android/ui/activities/LoginSmsActivity.kt
package com.instar.frontend_android.ui.activities

import android.content.Context
import android.os.Bundle
import android.text.InputType
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.Button
import android.widget.EditText
import android.widget.ImageButton
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.instar.frontend_android.R
import com.instar.frontend_android.databinding.EdittextLoginBinding
import com.instar.frontend_android.databinding.ActivityLoginSmsBinding
import com.instar.frontend_android.ui.customviews.ViewEditText
import com.instar.frontend_android.ui.customviews.ViewEffect


class LoginSmsActivity : AppCompatActivity() {
    private lateinit var binding: ActivityLoginSmsBinding

    private lateinit var imageBack: ImageButton
    private lateinit var btnText: TextView
    private lateinit var title: TextView
    private lateinit var textNote: TextView
    private lateinit var layout: View
    private lateinit var layoutSms: EdittextLoginBinding
    private lateinit var SmsText: EditText
    private lateinit var labelSms: TextView
    private lateinit var btnEyes: ImageButton
    private lateinit var btnNewSms: TextView
    private lateinit var btnSms: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityLoginSmsBinding.inflate(layoutInflater)
        setContentView(binding.root)

        imageBack = binding.imageBack
        layout = binding.layout
        layoutSms = binding.sms
        SmsText = layoutSms.editText
        labelSms = layoutSms.textView
        btnEyes = layoutSms.btnImage
        title = binding.smsTitle.Title
        textNote = binding.smsTitle.textNote
        btnText = binding.smsTitle.btnText
        btnNewSms = binding.btnNewSms
        btnSms = binding.btnSms

        loadActivity()
        initView()
    }

    private fun loadActivity() {
        title.text = "Xác nhận tài khoản"
        textNote.text = "Chúng tôi đã gửi mã đến email của bạn. Hãy nhập mã đó để xác nhận tài khoản"
        btnText.text = "Không thể xác nhận tài khoản?"
        labelSms.text = "Nhập mã"
        btnNewSms.text = "Gửi lại mã"
        SmsText.hint = "Nhập mã"
        btnSms.text = "Tiếp tục"
        SmsText.inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
        btnEyes.setBackgroundResource(R.drawable.ic_instagram_eyes_off)
    }

    private fun initView() {
        val viewEditText = ViewEditText()
        viewEditText.EditTextEyes(layoutSms.root, SmsText, labelSms, btnEyes)
        ViewEffect.ViewText(btnText)
        btnText.setOnClickListener { }

        layout.setOnClickListener {
            val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.hideSoftInputFromWindow(it.windowToken, 0)
            SmsText.clearFocus()
            if (SmsText.text.toString().isEmpty()) {
                setSms()
            }
        }

        imageBack.setOnClickListener { finish() }

        effectClick()
    }

    private fun setSms() {
        labelSms.visibility = View.GONE
        layoutSms.root.background = getDrawable(R.drawable.border_component_login_dow)
        SmsText.hint = "Nhập mã"
    }

    private fun effectClick() {
        ViewEffect.ViewText(btnNewSms)
        ViewEffect.ImageBack(imageBack)
    }
}